### DISCUSSION TITLE
Config contexts in config template

### DISCUSSION THREAD

**Discussion by Mailstorm-ctrl** [netbox-community/netbox#15407] [Help Wanted!] [answered]
I can't seem to figure this out. It seems possible from what I read but nothing I try works.

Suppose I have a config context such as:

```
{
    "aaa_servers": [
        "MS-NAPA-AAA-01",
        "MS-NAPA-AAA-02"
    ],
    "ntp_servers": [
        "MS-NAPA-DC-01",
        "MS-NAPA-DC-02",
        "MS-NA-NTP1"
    ]
}
```
This config context is linked to a region. This region also has devices that are linked to it. Now when I go to a device in that region and click on "Config Context" I can clearly see the above JSON value in rendered context and source contexts. I'd like to be able to use this data in the render config, otherwise known as the config template in netbox.

It seems no matter what I do I just can't find out how to do it. I even tried using nbshell and doing `print([f"{f.name} - {type(f).__name__}" for f in Device.objects.filter(name="devicenamehere")[0]._meta.get_fields()])` to get all attributes of my device. There's only two attributes in there I thought it might be but they both return `None` inside the nbshell.

So yeah...how do I use those contexts inside the config template?

**Comment by candlerb** [comment, marked as answer]
See #15386, #11937.

Also, in nbshell, `help(Device)` will give you all the methods and properties on the object. If you search for "context" you'll find

```
 |  ----------------------------------------------------------------------
 |  Methods inherited from extras.models.configs.ConfigContextModel:
 |
 |  get_config_context(self)
 |      Compile all config data, overwriting lower-weight values with higher-weight values where a collision occurs.
 |      Return the rendered configuration context for a device or VM.
 |
 |  ----------------------------------------------------------------------
```

**Reply by Mailstorm-ctrl** [reply]
Huh, not sure why I wasn't getting this when I searched for this question. But thank you!
